extensions-contrib/influx-extensions/src/test/java/io/druid/data/input/influx/InfluxParserTest.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.data.input.influx;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import io.druid.java.util.common.Pair;
+import io.druid.java.util.common.parsers.ParseException;
+import io.druid.java.util.common.parsers.Parser;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.isA;
+
+@RunWith(JUnitParamsRunner.class)
+public class InfluxParserTest
+{
+  private String name;
+  private String input;
+  private Map<String, Object> expected;
+
+  private static Object[] testCase(String name, String input, Parsed expected)
+  {
+    return Lists.newArrayList(name, input, expected).toArray();
+  }
+
+
+  public Object[] testData()
+  {
+    return Lists.newArrayList(
+        testCase(
+            "real sample",
+            "cpu,host=foo.bar.baz,region=us-east-1,application=echo pct_idle=99.3,pct_user=88.8,m1_load=2i 1465839830100400200",
+            Parsed.row("cpu", 1465839830100L)
+                  .with("host", "foo.bar.baz")
+                  .with("region", "us-east-1")
+                  .with("application", "echo")
+                  .with("pct_idle", 99.3)
+                  .with("pct_user", 88.8)
+                  .with("m1_load", 2L)
+        ),
+        testCase(
+            "negative timestamp",
+            "foo,region=us-east-1,host=127.0.0.1 m=1.0,n=3.0,o=500i -123456789",
+            Parsed.row("foo", -123L)
+                  .with("region", "us-east-1")
+                  .with("host", "127.0.0.1")
+                  .with("m", 1.0)
+                  .with("n", 3.0)
+                  .with("o", 500L)
+        ),
+        testCase(
+            "truncated timestamp",
+            "foo,region=us-east-1,host=127.0.0.1 m=1.0,n=3.0,o=500i 123",
```

Comment:
I think a timestamp like "500i 123" should throw a ParseException instead of being treated as 0, can you add a check on the truncated portion to make sure that it's also numeric?

Reply:
The 500i there is the (integer) value for the measurement named `o` and `123` is a valid timestamp. Invalid timestamps are caught by the parser since a valid line has to end with a number, newline or EOF.

Reply:
Ah okay, sorry, misread it, I see you have a case already for that type of failure in testParseFailures